src/bidi.jl

```diff
@@ -0,0 +1,31 @@
+import Base: size, getindex, setindex!
+
+#z indices are flipped for even-numbered timepoints
+struct BidiImageArray{T} <: AbstractArray{T,4}
+    A::AbstractArray{T,4}
+    z_size::Int
+end
+
+BidiImageArray(A::AbstractArray{T,4}) where {T} = BidiImageArray(A, size(A,3))
+
+size(B::BidiImageArray) = size(B.A)
+
+Base.IndexStyle(::Type{<:BidiImageArray}) = IndexCartesian()
+
+function getindex(B::BidiImageArray{T}, I::Vararg{Int, 4}) where {T}
+    t_ind = I[4]
+    if isodd(t_ind)
```

Comment:
Rather than a branch here, I recommend
```julia
iz = ifelse(isodd(t_ind), I[3], B.z_size - I[3] + 1)
return getindex(B.A, I[1], I[2], iz, t_ind)
```
The reasoning is well explained in [this famous stackoverflow answer](https://stackoverflow.com/questions/11227809/why-is-it-faster-to-process-a-sorted-array-than-an-unsorted-array) (and `ifelse` generates a [`select` instruction](https://llvm.org/docs/LangRef.html#select-instruction) with no branching). (There are times where the branch is necessary or superior, because `ifelse` requires that you compute both arguments no matter what.) In addition to avoiding branch-prediction failure, avoiding a branch here will also allow SIMD vectorization for operations performed on a `BidiImageArray`.

Demo:
```julia
julia> function foo1(A)
           n = 0
           @inbounds for a in A
               if a > 0.5
                   n += 1
               else
                   n -= 1
               end
           end
           n
       end
foo1 (generic function with 1 method)

julia> function foo2(A)
           n = 0
           @inbounds for a in A
               n += ifelse(a > 0.5, 1, -1)
           end
           n
       end
foo2 (generic function with 1 method)

julia> using BenchmarkTools

julia> A = rand(10^7);

julia> @btime foo1($A) seconds=1
  8.078 ms (0 allocations: 0 bytes)
3400

julia> @btime foo2($A) seconds=1
  4.724 ms (0 allocations: 0 bytes)
3400
```